libtenzir/builtins/functions/duration.cpp

```diff
@@ -76,8 +76,83 @@ class plugin final : public function_plugin {
     });
   }
 };
+
+class subduration_plugin final : public function_plugin {
+public:
+  subduration_plugin() = default;
+
+  subduration_plugin(std::string name, tenzir::duration unit)
+    : name_{std::move(name)}, unit_{unit} {
+  }
+
+  auto name() const -> std::string override {
+    return name_;
+  }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    TRY(argument_parser2::function(name())
+          .positional("x", expr, "duration")
+          .parse(inv, ctx));
+    return function_use::make(
+      [this, expr = std::move(expr)](evaluator eval, session ctx) -> series {
+        auto b = uint64_type::make_arrow_builder(arrow::default_memory_pool());
+        check(b->Reserve(eval.length()));
+        for (auto& arg : eval(expr)) {
+          match(
+            *arg.array,
+            [&](const arrow::NullArray& arg) {
+              check(b->AppendNulls(arg.length()));
+            },
+            [&](const arrow::DurationArray& arg) {
```

Comment:
These functions are nice! Should they also work if the input is a numeric array? I recently wrote `ts = from_epoch(x * 1ms)`, and I thought that `ts =  x.milliseconds().from_epoch()` would've read much nicer.

Reply:
Added these and renamed original ones to `count_*`.